doc/source/language/user_functions.rst

```diff
@@ -0,0 +1,339 @@
+.. _user_functions:
+
+**KerboScript** User Functions
+==============================
+
+.. contents::
+    :local:
+    :depth: 2
+
+This page covers functions created by you, the user of kerboscript,
+rather than the built-in functions provided by kOS.
+
+Help for the new user - What is a Function?
+-------------------------------------------
+
+    In programming terminology, there is a commonly used feature of
+    many programming languages that works as follows:
+
+    - 1. Create a chunk of program instructions that you don't intend to execute YET.
+    - 2. Later, when executing other parts of the program, do the following:
+       - A. Remember the current location in the program.
+       - B. Jump to the previously created chunk of code from (1) above.
+       - C. Run the instructions there.
+       - D. Return to where you remembered from (A) and continue from there.
+
+    This feature goes by many different names, with slightly different
+    precise meanings: *Subroutines*, *Procedures*, *Functions*, etc.
+    For the purposes of kerboscript, we will refer to all uses of this
+    feature with the term *Function*, whether it *technically* fits the
+    mathematical definition of a "function" or not.
+
+.. _declare function:
+
+``DECLARE FUNCTION``
+--------------------
+
+In kerboscript, you can make your own user functions using the
+DECLARE FUNCTION command, which has syntax as follows:
+
+  ``declare function`` *identifier* ``{`` *statements* ``}`` *optional dot (.)*
+
+example::
+
+    // Print the string you pass in, in one of the 4 corners
+    // of the terminal:
+    //   mode = 1 for upper-left, 2 for upper-right, 3
+    //          for lower-left, and 4 for lower-right:
+    //
+    declare function print_corner {
+      declare parameter mode.
+      declare parameter text.
+
+      declare row.
+      declare col.
+
+      if mode = 1 or mode = 3 {
+        set col to 0.
+      } else {
+        set col to terminal:width - text:length.
+      }.
+      if mode = 1 or mode = 2 {
+        set row to 0.
+      } else {
+        set row to terminal:height - 1.
+      }.
+
+      print text at (col, row).
+    }.
+
+    // An example of calling it:
+    
+    print_corner(4,"That's me in the corner").
+
+``Declare function`` can appear anywhere in a kerboscript program,
+and once its been "parsed" by the compiler, the function can be called
+from anywhere in the program.  
+
+The best design pattern is probably to create your library of function
```

Comment:
I think it should error as well. i cannot think of a good reason actually want to replace a function or a kOS binding